etcd/v1alpha1/types_pacemakercluster.go

```diff
@@ -0,0 +1,489 @@
+	// +optional
+	OnlineStatus NodeOnlineStatusType `json:"onlineStatus,omitempty"`
+
+	// mode indicates if the node is in active or standby mode
+	// NodeModeType can be one of the following values:
+	// - Active - the node is in active mode
+	// - Standby - the node is in standby mode
+	// When present, it must be a valid NodeModeType.
+	// When not present, the node mode is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +optional
+	Mode NodeModeType `json:"mode,omitempty"`
+}
+
+// PacemakerResourceStatus represents the status of a single resource in the Pacemaker cluster
+type PacemakerResourceStatus struct {
+	// name is the name of the resource
+	// When present, it must be a valid string between 1 and 256 characters long.
+	// When not present, the resource name is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// resourceAgent is the resource agent type (e.g., "ocf:heartbeat:IPaddr2", "systemd:kubelet")
+	// When present, it must be a valid string between 1 and 256 characters long.
+	// When not present, the resource agent type is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +optional
+	ResourceAgent string `json:"resourceAgent,omitempty"`
+
+	// role is the current role of the resource
+	// ResourceRoleType can be one of the following values:
+	// - Started - the resource is started
+	// - Stopped - the resource is stopped
+	// We don't use promoted and unpromoted, so resources in those roles would omit the role field.
+	// When present, it must be a valid ResourceRoleType.
+	// When not present, the resource role is unknown (or an unsupported type like promoted or unpromoted). This likely indicates that there is an error parsing the raw XML output.
+	// +optional
+	Role ResourceRoleType `json:"role,omitempty"`
+
+	// activeStatus indicates if the resource is active or inactive
+	// ResourceActiveStatusType can be one of the following values:
+	// - Active - the resource is active
+	// - Inactive - the resource is inactive
+	// When present, it must be a valid ResourceActiveStatusType.
+	// When not present, the active status is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +optional
+	ActiveStatus ResourceActiveStatusType `json:"activeStatus,omitempty"`
+
+	// node is the node where the resource is running
+	// When present, it must be a valid string between 1 and 256 characters long.
+	// When not present, the node name is unknown. This likely indicates that there is an error parsing the raw XML output.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +optional
+	Node string `json:"node,omitempty"`
```

Comment:
Do you support clone resources?  If so, those can run on multiple nodes at the same time in which case making this some sort of list type would make more sense to me.  Also if you care about clones, keep in mind that the name of the primitive resource being cloned is not unique.